src/middlewared/middlewared/plugins/catalogs_linux/update.py

```diff
@@ -193,6 +193,27 @@ async def common_validation(self, catalog, schema, data):
 
         verrors.check()
 
+    @private
+    async def dataset_mounted(self):
+        if k8s_dataset := (await self.middleware.call('kubernetes.config'))['dataset']:
+            if catalog_ds := await self.middleware.call(
+                'zfs.dataset.query', [['id', '=', os.path.join(k8s_dataset, 'catalogs')]], {
+                    'extra': {'properties': ['mounted']}
```

Comment:
Why `zfs.dataset.query`? This is expensive even if you just do `properties: ['mounted']` since it retrieves all children by default.

Why not check `filesystem.mount_info`?

Reply:
There are no children of this dataset but yup i can use that one to retrieve mount related data